src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -102,19 +108,33 @@ internal static bool TryReadMemoryValueFromFile(string path, out ulong result)
             return false;
         }
 
+        /// <summary>Find the cgroup version in use on the system.</summary>
+        /// <returns>The cgroup version.</returns>
+        private static CGroupVersion FindCGroupVersion()
+        {
+            var driveInfo = new DriveInfo(SysFsCgroupFileSystemPath);
+            switch (driveInfo.DriveFormat)
+            {
+                case "cgroup2fs": return CGroupVersion.CGroup2;
+                case "tmpfs": return CGroupVersion.CGroup1;
+                default: return CGroupVersion.None;
+            }
+        }
```

Comment:
FWIW, this could be condensed slightly to:
```C#
private static CGroupVersion FindCGroupVersion() =>
    new DriveInfo(SysFsCgroupFileSystemPath).DriveFormat switch
    {
        "cgroup2fs" => CGroupVersion.CGroup2,
        "tmpfs" => CGroupVersion.CGroup1,
        _ => CGroupVersion.None
    };
```

Reply:
Thanks! Fixed.

Reply:
Sorry, I have had to undo a part of this because `new DriveInfo(SysFsCgroupFileSystemPath)` throws exceptions on non-Linux OSes and it needs to be captured and handled.